### ISSUE TITLE
No. of clusters optional parameter in getclusters(data, cluster_number) not working

### ISSUE THREAD

**Issue by kdamar3** [shudima/dimas-kmeans#1]
Hi, i tried giving no of clusters as getClusters(data,5). 
But, it still clusters by default settings for a dummy dataset of 15 arrays as 3 clusters. Which is also have to be around 6 or 7 clusters according to specification as half the size rather than 3.

**Comment by shudima** [comment]
Hi, 
Can you share your code?
Btw, the default number of clusters is defined by: sqrt(n/2))
Also, in order to use the "numberOfClusters" option you should do:

`var clusters = kmeans.getClusters(data, {numberOfClusters:7});`

**Comment by kdamar3** [comment]
Hi Shudima, Thanks for reply, 

Here is the jsfiddle of code where i used dimas-kmeans: https://jsfiddle.net/data_x/nk2mhhtc/18/

I created a  dataset with 15 arrays.  At line 107, i am getting clustered data from getClusters(data, {numberOfClusters:5}); Upon clicking the button cluster, you get new cluster data with each cluster bearing different color. 

 The result i got from console for getcluster(data,{numberOfClusters:5}) and the visual output is all the time only 3 clustered arrays. though the console displays 5 different arrays, only first 3 have clustered data and remaining last 2 are 0. You can try giving different number of clusters, i tried with 7 too. its the same case. all the time, only first 3 arrays have clustered data and remaining others 0.

Sorry, yeah, the default is root of half of size. But when i change no. of clusters option, then it is not working as expected. 

Thank you,

**Comment by shudima** [comment]
I can see from the jsfiddle that it is working as expected. Please note that some of the clusters might be empty so you won't see them in the viz, print the `clustered_data.length` and you should see the correct number of clusters

**Comment by kdamar3** [comment]
Yes, Please look at the image http://imgur.com/BXLrtha. On right side, console message says length as 5 and displays the clusters as 5. But, all the time, only first 3 arrays are distributed with data and last arrays are 0 with no clustered data. 

The same was displayed in viz. viz display 5 colors if data was in 5 arrays.  

I thought when user input no. of clusters, the data was distributed over all these clusters.  Please let me know if my understanding of kmeans clustering algorithm is wrong.

Thank you,

**Comment by shudima** [comment]
K-Means can create empty clusters.
Please take a look at this example 
[http://user.ceng.metu.edu.tr/~tcan/ceng465_f1314/Schedule/KMeansEmpty.html](url)
